File: core/src/main/java/pcb/uwu/units/composite/mechanics/ForceUnit.java
package pcb.uwu.units.composite.mechanics;

import pcb.uwu.core.CompositeUnit;
import pcb.uwu.core.UnitCounter;
import pcb.uwu.units.quantity.LengthUnit;
import pcb.uwu.units.quantity.MassUnit;
import pcb.uwu.units.quantity.TimeUnit;

public class ForceUnit extends CompositeUnit {

	// region base constructor

	public ForceUnit(MassUnit mass, LengthUnit length, TimeUnit time) {
		super(new UnitCounter()
				.major(mass)
				.major(length)
				.minor(time, 2));
	}

	// endregion

	// region composite constructors

	public ForceUnit(MassUnit massUnit, AccelerationUnit accelerationUnit) {
		super(new UnitCounter()
				.major(massUnit)
				.major(accelerationUnit.getUnitCounter()));
	}

	// endregion

}

File: dimension/src/main/java/pcb/uwu/unit/derived/mechanics/ForceUnit.kt
package pcb.uwu.unit.derived.mechanics

import pcb.uwu.core.CompositeUnit
import pcb.uwu.core.UnitCounter
import pcb.uwu.unit.base.LengthUnit
import pcb.uwu.unit.base.MassUnit
import pcb.uwu.unit.base.TimeUnit
import pcb.uwu.unit.derived.geometry.AreaUnit
import pcb.uwu.unit.derived.termodynamics.EnergyUnit
import pcb.uwu.unit.derived.termodynamics.PressureUnit

open class ForceUnit : CompositeUnit
{
    constructor(massUnit: MassUnit,
                lengthUnit: LengthUnit,
                timeUnit: TimeUnit)
            : super(UnitCounter()
                        .major(massUnit)
                        .major(lengthUnit)
                        .minor(timeUnit)
                        .minor(timeUnit))

    constructor(massUnit: MassUnit,
                accelerationUnit: AccelerationUnit)
            : super(UnitCounter()
                        .major(massUnit)
                        .major(accelerationUnit))

    constructor(energyUnit: EnergyUnit,
                lengthUnit: LengthUnit)
            : super(UnitCounter()
                        .major(energyUnit)
                        .minor(lengthUnit))

    constructor(pressureUnit: PressureUnit,
                areaUnit: AreaUnit)
            : super(UnitCounter()
                        .major(pressureUnit)
                        .minor(areaUnit))
}